File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let split (logLine: string) = 
    match logLine.Split ':' with
    | [|level; message|] -> (level.Trim([|'['; ']'|]).ToLower(), message.Trim())
    | _ -> failwith $"It's not a {nameof logLine} = \"{logLine}\"."

let message (logLine: string): string =
    let (_, message) = split logLine
    message

let logLevel(logLine: string): string =
    let (level, _) = split logLine
    level

let reformat(logLine: string): string =
    let (level, message) = split logLine
    $"{message} ({level})"

File: solutions/csharp/log-levels/1/LogLevels.cs
using System;

static class LogLine
{
    static (string LogLevel, string Message) Split(string logLine) {
        var line = logLine.Split(':');
        return (line[0][1..^1].ToLower(), line[1].Trim());
    }

    public static string Message(string logLine) => Split(logLine).Message;

    public static string LogLevel(string logLine) => Split(logLine).LogLevel;

    public static string Reformat(string logLine) {
        (string LogLevel, string Message) = Split(logLine);
        return $"{Message} ({LogLevel})";
    }
}
